.github/workflows/codespeed.yml

```diff
@@ -0,0 +1,95 @@
+name: Codespeed Benchmarks
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  # Special job which automatically cancels old runs for the same branch, prevents runs for the
+  # same file set which has already passed, etc.
+  pre_job:
+    name: Skip Duplicate Jobs Pre Job
+    runs-on: ubuntu-latest
+    permissions:
+      actions: write  # Needed for skip-duplicate-jobs job
+      contents: read
+    outputs:
+      should_skip: ${{ steps.skip_check.outputs.should_skip }}
+    steps:
+      - id: skip_check
+        uses: fkirc/skip-duplicate-actions@9d116fa7e55f295019cfab7e3ab72b478bcf7fdd # v4.0.0
+        with:
+          cancel_others: 'true'
+          github_token: ${{ github.token }}
+
+  codespeed-micro-benchmarks:
+    runs-on: ubuntu-latest
+    needs: pre_job
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version:
+          - "2.7.17"
+          - "3.5.9"
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v3
+        with:
+          submodules: true
+
+      - name: Setup Python
+        uses: actions/setup-python@v4
+        id: setup-python
+        with:
+          python-version: ${{ matrix.python-version }}
+
+      - name: Install Deps
+        env:
+          TOX_VERSION: 3.20.1
+          TOX_GH_ACTIONS_VERSION: 2.9.1
+        run: |
+          python -m pip install --upgrade pip
+          pip install "tox==$TOX_VERSION" "tox-gh-actions==$TOX_GH_ACTIONS_VERSION" "cffi==1.12.3" "udatetime==0.0.16" "requests"
+          # Needed for snappy library
+          sudo apt-get update
+          sudo apt-get install -y libsnappy-dev
+
+      - name: Run Benchmarks
+        run: tox -e micro-benchmarks
+
+      - name: Process and submit results to CodeSpeed
+        if: (github.ref == 'refs/heads/master' || github.head_ref == 'master')
+        env:
+          PYTHONPATH: .
+          CODESPEED_URL: "https://scalyr-agent-codespeed.herokuapp.com/"
+          CODESPEED_PROJECT: "scalyr-agent-2-microbenchmarks"
+          CODESPEED_EXECUTABLE: Python ${{ steps.setup-python.outputs.python-version }}
+          CODESPEED_ENVIRONMENT: GitHub Hosted Action Runner
+          CODESPEED_AUTH: ${{ secrets.CODESPEED_AUTH }}
+        run: python benchmarks/scripts/send_microbenchmarks_data_to_codespeed.py --data-path="benchmark_results/*.json" --debug
```

Comment:
Per comment on Slack / Zoom the other day, we have multiple sets of benchmarks tracked in CodeSpeed:

1. micro benchmarks (what you have migrated here)
2. agent process level "end to end" benchmarks (where we run complete agent process with various different configurations and record various resource utilization metrics)

Do you plan to migrate 2) as part of this PR or as part of a different one?

Reply:
@Kami I am migrating 2) as part of CT-577. I broke them into separate tasks to try and keep PR sizes down (among other reasons).